File: HYPERLEDGER/bfn-network-apps/businesslibrary/lib/stellar/Account.dart
import 'package:businesslibrary/stellar/Balance.dart';
import 'package:businesslibrary/stellar/Data.dart';
import 'package:businesslibrary/stellar/Flags.dart';
import 'package:businesslibrary/stellar/Links.dart';
import 'package:businesslibrary/stellar/Signer.dart';
import 'package:businesslibrary/stellar/Thresholds.dart';
import 'package:json_annotation/json_annotation.dart';

part 'Account.g.dart';

@JsonSerializable()
class Account extends Object with _$AccountSerializerMixin {
  Links links;
  String id;
  String paging_token;
  String account_id;
  String sequence;
  int subentry_count;
  Thresholds thresholds;
  Flags flags;
  List<Signer> signers;
  Data data;
  List<Balance> balances;

  Account(
      {this.links,
      this.id,
      this.paging_token,
      this.account_id,
      this.sequence,
      this.subentry_count,
      this.thresholds,
      this.flags,
      this.signers,
      this.data,
      this.balances});

  factory Account.fromJson(Map<String, dynamic> json) =>
      _$AccountFromJson(json);
}

File: HYPERLEDGER/bfn-network-chaincode/bfnwebapi-node/src/server/models/account.ts
import { JsonObject, JsonProperty } from 'json2typescript';

@JsonObject("Balance")
export class Balance {
  @JsonProperty()
  public balance: string | undefined;
  @JsonProperty()
  public asset_type: string | undefined;
}
@JsonObject("Account")
export class Account {
  @JsonProperty()
  public id: string | undefined;
  @JsonProperty()
  public paging_token: string | undefined;
  @JsonProperty()
  public account_id: string | undefined;
  @JsonProperty()
  public sequence: string | undefined;
  @JsonProperty()
  public balances: Balance[] | undefined;
}
